internal/dataplane/translator/translate_upstreams.go

```diff
@@ -310,15 +323,26 @@ func getEndpoints(
 				// In most cases consumers should interpret this unknown state as ready.
 				// Field Ready has the same semantic as Endpoints from CoreV1 in Addresses.
 				// https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/#conditions
-				if endpoint.Conditions.Ready != nil && !*endpoint.Conditions.Ready {
-					continue
+
+				isTerminating := endpoint.Conditions.Terminating != nil && *endpoint.Conditions.Terminating
+				isReady := endpoint.Conditions.Ready == nil || *endpoint.Conditions.Ready
+
+				// Skip endpoints that are not ready
+				if !isReady {
+					// Only include terminating endpoints if the feature gate is enabled
+					if !isTerminating || !enableStickySessionsTerminatingEndpoints {
+						continue
+					}
 				}
+
 				// One address per endpoint is rather expected (allowing multiple is due to historical reasons)
 				// read more https://github.com/kubernetes/kubernetes/issues/106267#issuecomment-978770401.
 				// These are all assumed to be fungible and clients may choose to only use the first element.
 				upstreamServer := util.Endpoint{
 					Address: endpoint.Addresses[0],
 					Port:    upstreamPort,
+					// Only mark as terminating if the feature gate is enabled
+					Terminating: isTerminating && enableStickySessionsTerminatingEndpoints,
```

Comment:
We have already skipped the terminating endpoints when sticky session is not enabled, so is it necessary to add the `&& enableStickySessionsTerminatingEndpoints` here?

Reply:
I am considering that if there are some unconsumed endpoints when we modify the configuration of KIC, then this situation may occur.